src/components/ReplaceServerDetails.js

```diff
@@ -0,0 +1,321 @@
+      else {
+        this.allInputsStatus[key] = INPUT_STATUS.INVALID;
+      }
+    }
+    this.setState({isFormValid: this.isFormTextInputValid()});
+  }
+
+  handleOsPasswordChange = (e) => {
+    const password = e.target.value;
+    this.setState({osInstallPassword: password}, () => {
+      this.setState({isFormValid: this.isFormTextInputValid()});
+    });
+  }
+
+  handleInstallOsCheck = () => {
+    this.setState({
+      isInstallOsChecked: !this.state.isInstallOsChecked }, () => {
+      this.setState({isFormValid: this.isFormTextInputValid()});
+    });
+  }
+
+  handleWipeDiskCheck = () => {
+    this.setState({isWipeDiskChecked: !this.state.isWipeDiskChecked});
+  }
+
+  handleSelectAvailableServer = (serverId) => {
+    //TODO need to find a way to show the details of the available server
+    //selected
+    this.setState({selectedServerId: serverId});
+    let allAvailableServers = [];
+    if(this.props.autoServers && this.props.autoServers.length > 0) {
+      allAvailableServers = this.props.autoServers;
+    }
+    if(this.props.manualServers && this.props.manualServers.length > 0) {
+      allAvailableServers = allAvailableServers.concat(this.props.manualServers);
+    }
+
+    let theServer = allAvailableServers.find(server => server.id === serverId);
+    let newJSData = {};
+    if(theServer) {
+      newJSData['ilo-ip'] = theServer['ilo-ip'] ? theServer['ilo-ip'] : '';
+      newJSData['ilo-user'] = theServer['ilo-user'] ? theServer['ilo-user'] : '';
+      newJSData['ilo-password'] = theServer['ilo-password'] ? theServer['ilo-password'] : '';
+      newJSData['mac-addr'] = theServer['mac-addr'] ? theServer['mac-addr'] : '';
+      //record server uid from ov or id from sm, so it can be filtered out for available
+      //servers later
+      newJSData['uid'] =
+        theServer['uid'] ? theServer['uid'] : (theServer['id'] ? theServer['id'] : '');
+      this.setState({
+        replaceData: fromJS(newJSData),
+      }, () => {
+        this.updateAutoFillFormValidity(newJSData);
+      });
+    }
+  }
+
+  renderInput(name, type, isRequired, title, validate) {
+    return (
+      <InputLine
+        isRequired={isRequired} inputName={name} inputType={type} label={title}
+        inputValidate={validate}
+        inputValue={this.state.replaceData.get(name) ? this.state.replaceData.get(name) : ''}
+        inputAction={this.handleInputChange} updateFormValidity={this.updateFormValidity}/>
+    );
+  }
+
+  renderText(title, value) {
+    return (
+      <div className='detail-line'>
+        <div className='detail-heading'>{translate(title)}</div>
+        <div className='info-body'>{value}</div>
+      </div>
+    );
+  }
+
+  renderOsPasswordInput() {
+    return (
+      <InputLine
+        isRequired={this.state.isInstallOsChecked} inputName={'osInstallPassword'}
+        inputType={'password'} label={'server.pass.prompt'}
+        inputValue={this.state.osInstallPassword ? this.state.osInstallPassword : ''}
```

Comment:
use `this.state.osInstallPassword || ''`

Reply:
will update